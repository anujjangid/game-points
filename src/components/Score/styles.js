import styled from 'styled-components';

export const TabelHead = styled.div`
  background: grey;
  display: flex;
  color: #fff;
  text-transform: uppercase;
  div {
    padding: 10px;
    box-sizing: border-box;
    text-align: center;
  }
`;

export const List = styled.section`
  background: #fff;
  display: flex;
  color: #333;
  text-transform: uppercase;
  border-bottom: 1px solid #333;
  margin-bottom: 10px;
  border-top: 1px solid #333;
  width: 88%;
  margin: 10px auto;
  border-radius: 10px;
  div {
    padding: 12px;
    box-sizing: border-box;
    text-align: center;
    font-size: 22px;
  }
`;

export const ScoreFooter = styled.footer`
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  margin-top: auto;
  background: #fff;
`;

export const ScoreItem = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 10px;
  font-size: 24px;
`;

export const Trigger = styled.button`
  font-size: 18px;
  cursor: pointer;
  border: none;
  padding: 10px;
  display: flex;
  text-transform: uppercase;
  letter-spacing: 1px;
  width: 100%;
  justify-content: center;
  transition: all 0.3s;
  background: #cb4e4e;
  color: #fff;
  box-shadow: 0 6px #ab3c3c;
  border-radius: 5px;
  margin: 10px 0 10px 0;
  &:hover {
    box-shadow: 0 4px #ab3c3c;
    top: 2px;
  }
`;

export const ScoreBlock = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  height: 100%;
  border-left: 1px solid #8acdd2;
  background: gray;
  @media only screen and (min-width: 768px) {
    border: none;
  }
`;
