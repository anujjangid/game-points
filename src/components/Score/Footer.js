import React, { useContext } from 'react';

import { getBonus, getTotal } from '../../utils';
import { BonusText, TotalText } from '../../Constants';
import { CounterContext } from '../../reducer';
import { ScoreFooter, ScoreItem } from './styles';

const Footer = props => {
  const { state } = useContext(CounterContext);

  return (
    <ScoreFooter>
      <ScoreItem>
        <span>{BonusText}</span>
        <span>{(state && state.items && getBonus(state.items)) || 0}</span>
      </ScoreItem>
      <ScoreItem>
        <span>{TotalText}</span>
        <span>{(state && state.items && getTotal(state.items)) || 0}</span>
      </ScoreItem>
    </ScoreFooter>
  );
};

export default Footer;
