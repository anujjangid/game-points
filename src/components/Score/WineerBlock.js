import React from 'react';
import PropTypes from 'prop-types';

import { Column } from '../../styles';
import { List } from './styles';

const WineerBlock = props => {
  const {
    data: { price, game, count }
  } = props;

  return (
    <React.Fragment>
      <List key={price + game}>
        <Column span="4">{game}</Column>
        <Column span="4">{count}</Column>
        <Column span="4">{price}</Column>
      </List>
    </React.Fragment>
  );
};

export default WineerBlock;

WineerBlock.propTypes = {
  data: PropTypes.shape({
    bonus: PropTypes.number,
    count: PropTypes.number.isRequired,
    game: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired
  })
};
