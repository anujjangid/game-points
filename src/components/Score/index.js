import React, { useContext } from 'react';

import HeaderBar from '../HeaderBar';
import { PlayerTitle, ItemHeading, ItemQty, ItemScore } from '../../Constants';
import Footer from './Footer';
import WineerBlock from './WineerBlock';
import { CounterContext } from '../../reducer';
import { TabelHead, Trigger, ScoreBlock } from './styles';
import { Column } from '../../styles';
import { NewGame } from '../../Constants';

const Score = () => {
  const { state, dispatch } = useContext(CounterContext);

  return (
    <Column span="4">
      <ScoreBlock>
        <HeaderBar title={PlayerTitle} />
        <TabelHead>
          <Column span="4">{ItemHeading}</Column>
          <Column span="4">{ItemQty}</Column>
          <Column span="4">{ItemScore}</Column>
        </TabelHead>

        {state &&
          state.items &&
          Object.keys(state.items).map(function(key) {
            return (
              <WineerBlock
                key={state.items[key].price + state.items[key].game}
                data={state.items[key]}
              />
            );
          })}
        <Footer />
        <Trigger onClick={() => dispatch({ type: 'reset' })}>{NewGame}</Trigger>
      </ScoreBlock>
    </Column>
  );
};

export default Score;
