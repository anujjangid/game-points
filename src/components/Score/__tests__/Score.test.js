import React from 'react';
import { create } from 'react-test-renderer';
import Score from '../index';

describe('Test suite for Score Component', () => {
  // create a UI snapshot to indentify UI changes or design regressions

  it('render correctly', () => {
    const component = create(<Score />);
    expect(component.toJSON()).toMatchSnapshot();
    const rootInstance = component.root;
    // check heading
    const Heading = rootInstance.findByType('h4');
    expect(Heading.props.children).toBe('PLAYERS ITEMS');
  });
});
