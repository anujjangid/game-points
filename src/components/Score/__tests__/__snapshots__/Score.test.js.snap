// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Score Component render correctly 1`] = `
<div
  className="sc-dnqmqq jxUQYu"
  span="4"
>
  <div
    className="sc-bZQynM eQktHo"
  >
    <header
      className="sc-bdVaJa eQZKMo"
    >
      <h4>
        PLAYERS ITEMS
      </h4>
    </header>
    <div
      className="sc-bwzfXH bWxlZi"
    >
      <div
        className="sc-dnqmqq jxUQYu"
        span="4"
      >
        Item
      </div>
      <div
        className="sc-dnqmqq jxUQYu"
        span="4"
      >
        Qty
      </div>
      <div
        className="sc-dnqmqq jxUQYu"
        span="4"
      >
        Score
      </div>
    </div>
    <footer
      className="sc-bxivhb edvcqL"
    >
      <div
        className="sc-ifAKCX hxhCpK"
      >
        <span>
          Bonus
        </span>
        <span>
          0
        </span>
      </div>
      <div
        className="sc-ifAKCX hxhCpK"
      >
        <span>
          Total
        </span>
        <span>
          0
        </span>
      </div>
    </footer>
    <button
      className="sc-EHOje dLWkVi"
      onClick={[Function]}
    >
      New Game
    </button>
  </div>
</div>
`;
